datalake/src/main/java/com/sequenceiq/datalake/cm/ClouderaManagerRangerUtil.java

```diff
@@ -117,7 +124,12 @@ public void setAzureCloudIdentityMapping(String stackCrn, Map<String, String> az
             rolesResourceApi.updateRoleConfig(clusterName, rangerUserSyncRoleName, RANGER_SERVICE_NAME,
                     "Updating Azure Cloud Identity Mapping through Cloudbreak",
                     configList);
-            triggerRoleRefresh(client, clusterName, RANGER_SERVICE_NAME, rangerUserSyncRoleName);
+            if (isRoleRefreshNeeded(rolesResourceApi, clusterName, rangerUserSyncRoleName)) {
+                LOGGER.info("Role refresh required, trigerring role refresh");
+                triggerRoleRefresh(client, clusterName, RANGER_SERVICE_NAME, rangerUserSyncRoleName);
+            } else {
+                LOGGER.info("No role refresh required");
+            }
```

Comment:
`setAzureCloudIdentityMapping` is already a big method, could you refactor this part into a separate one, like `refreshRoleIfStale`